### ISSUE TITLE
Putt full project files

### ISSUE THREAD

**Issue by masoudline** [ckflight/unh-startracker#1]
Hi, my friend
thanks for your good thesis and project.
I'm a mechatronic engineering and I will use your project in my thesis from India but your code not complete and has many errors.
please send your full project to my gmail: mimsalehi69@gmail.com or put them into your github account.

thanks for your favor

**Comment by fsbr** [comment]
Hi, what errors and incomplete parts are you referring to?

**Comment by fsbr** [comment]
The files you need to run are in unh-startracker/analysisAndPlots/maps/

You need a directory with image files for CPM.py to run, but you should be able to run ST.py with the right dependencies installed to your computer. I built this project in Ubuntu 14.04, so I think it should be possible to find all the libraries you need using that operating system.

**Comment by masoudline** [comment]
when i want to run cmp.py these error show:
`
anwcs.c:1039:anwcs_open_wcslib: Wcslib support was not compiled in
anwcs.c:1061:anwcs_open_wcstools: WCStools support was not compiled in
Traceback (most recent call last):
  File "/usr/local/astrometry/bin/plotann.py", line 316, in <module>
    jobjs = get_annotations_for_wcs(wcs, opt)
  File "/usr/local/astrometry/bin/plotann.py", line 140, in get_annotations_for_wcs
    if not wcs.is_inside(r, d):
  File "/usr/local/astrometry/lib/python/astrometry/util/util.py", line 1218, in is_inside
    return _util.anwcs_t_is_inside(self, ra, dec)
TypeError: in method 'anwcs_t_is_inside', argument 1 of type 'struct anwcs_t *'
Traceback (most recent call last):
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 312, in <module>
    runCenterPixelMethod()  
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 295, in runCenterPixelMethod
    starList = getStarDict(wcsFile)
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 130, in getStarDict
    starDict = simplejson.loads(starDict)
  File "/usr/local/lib/python2.7/dist-packages/simplejson/__init__.py", line 516, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python2.7/dist-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/local/lib/python2.7/dist-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.scanner.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
`
that i think those error show because my dataset is'nt normal. can you send me a standard dataset?

**Comment by fsbr** [comment]
I'll look for the proper dataset tonight. I don't go to the same university where the study was done so it might be hard to find. Please remind me if you don't hear back from me in a day or two.  You need to point at a directory with a lot of images of the stars in it.  Then, it uses the astrometry program to ID which stars are in it. I guess the question for you is what does your dataset look like right now?

**Comment by masoudline** [comment]
do you have Whatsapp? or imo or snapchat or skype?
I want to be run cpm.py many errors occur!

this is my skype account:

https://join.skype.com/invite/iba19jShk5Nz

**Comment by fsbr** [comment]
Ckflight on wechat

**Comment by masoudline** [comment]
do you have facebook account? i cant setup wechat on my phone. thanks

**Comment by fsbr** [comment]
We can use Google Hangout what time zone are you in

**Comment by masoudline** [comment]
ok its very well. my google account masoudline@gmail.com it's 06:04 PM

**Comment by fsbr** [comment]
Okay can you let me know a few times that are working for you

**Comment by masoudline** [comment]
i can every time that you can.

**Comment by fsbr** [comment]
also you need the software from here 
http://astrometry.net/

**Comment by masoudline** [comment]
i make astrometry.net but some errors appear like this:
```
ERROR: Mismatch in the CFITSIO_SONAME value in the fitsio.h include file
that was used to build the CFITSIO library, and the value in the include file
that was used when compiling the application program:
   Version used to build the CFITSIO library   = 2
   Version included by the application program = 5

Fix this by recompiling and then relinking this application program 
with the CFITSIO library.
augment-xylist.c:935:augment_xylist Source extraction failed
 image2xy-files.c:47:image2xy_files Failed to open FITS input file /tmp/tmp.fits.1S9m8d
 ../util/cfitsutils.h:26:cfitserr could not open the named file
wcsFile ========= DSC_0031_NEF_embedded.wcs 
command is ['/usr/local/astrometry/bin/plotann.py', '--brightcat', '/home/roxo/unh-startracker-master/astrometry/catalogs/brightstars.fits', '/home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.wcs']
anwcs.c:917:getheader: Failed to open file /home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.wcs
anwcs.c:1061:anwcs_open_wcstools: WCStools support was not compiled in
Traceback (most recent call last):
  File "/usr/local/astrometry/bin/plotann.py", line 316, in <module>
    jobjs = get_annotations_for_wcs(wcs, opt)
  File "/usr/local/astrometry/bin/plotann.py", line 140, in get_annotations_for_wcs
    if not wcs.is_inside(r, d):
  File "/usr/local/astrometry/lib/python/astrometry/util/util.py", line 1218, in is_inside
    return _util.anwcs_t_is_inside(self, ra, dec)
TypeError: in method 'anwcs_t_is_inside', argument 1 of type 'struct anwcs_t *'
Traceback (most recent call last):
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 310, in <module>
    runCenterPixelMethod()  
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 293, in runCenterPixelMethod
    starList = getStarDict(wcsFile)
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 128, in getStarDict
    starDict = json.loads(starDict)
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

**Comment by masoudline** [comment]
i tried to run your project and setup astrometry.net but these errors show:
```
You must list at least one index in the config file (/usr/bin/../etc/astrometry.cfg)

See http://astrometry.net/use.html about how to get some index files.
---------------------------------------------------------------------

solve-field.c:518:run_engine engine failed.  Command that failed was:
  /usr/bin/astrometry-engine /home/roxo/unh-startracker-master/dataset//apod2.axy
 ioutils.c:567:run_command_get_outputs Command failed: return value 255
Traceback (most recent call last):
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 310, in <module>
    runCenterPixelMethod()  
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 290, in runCenterPixelMethod
    t0 = extractTimeFromImage(image)
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 33, in extractTimeFromImage
    imgTime = imgTime[36868].split()
TypeError: 'NoneType' object has no attribute '__getitem__'

Process finished with exit code 1

```

**Comment by fsbr** [comment]
What image type are you trying to manipulate

On Oct 11, 2017 5:18 PM, "masoudline" <notifications@github.com> wrote:

i tried to run your project and setup astrometry.net but these errors show:

You must list at least one index in the config file
(/usr/bin/../etc/astrometry.cfg)

See http://astrometry.net/use.html about how to get some index files.
---------------------------------------------------------------------

solve-field.c:518:run_engine engine failed.  Command that failed was:
  /usr/bin/astrometry-engine
/home/roxo/unh-startracker-master/dataset//apod2.axy
 ioutils.c:567:run_command_get_outputs Command failed: return value 255
Traceback (most recent call last):
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py",
line 310, in <module>
    runCenterPixelMethod()
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py",
line 290, in runCenterPixelMethod
    t0 = extractTimeFromImage(image)
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py",
line 33, in extractTimeFromImage
    imgTime = imgTime[36868].split()
TypeError: 'NoneType' object has no attribute '__getitem__'

Process finished with exit code 1


—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/fsbr/unh-startracker/issues/1#issuecomment-335951676>,
or mute the thread

**Comment by masoudline** [comment]
The jpg image at the maps folder

The image is recorded by the camera and sent to the project. Is this project anything other than this

**Comment by fsbr** [comment]
Do you have a copy of the code you're running on github

**Comment by masoudline** [comment]
I have not got it yet, but I'll be added in the respository soon. What are you think about those errors?

**Comment by fsbr** [comment]
Add to the repo it's hard to answer questions without seeing what you did.

**Comment by masoudline** [comment]
I did not do anything special Just installing and loading this photo

**Comment by masoudline** [comment]
i run script and add demo image from astrometry/demo/DSC_0031_NEF_embedded.jpg and add index-4116 and index-219 for detection.  when run CMP.py below line show at output terminal:
```
/usr/bin/python2.7 /home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py
ATTEMPTING TO ACCESS DATA DIR /home/roxo/unh-startracker-master/dataset/
['/home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.jpg']
Reading input file 1 of 1: "/home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.jpg"...
jpegtopnm: WRITING PPM FILE
Read file stdin: 3872 x 2592 pixels x 1 color(s); maxval 255
Using 8-bit output
Extracting sources...
Downsampling by 2...
simplexy: found 2197 sources.
Solving...
Reading file "/home/roxo/unh-startracker-master/dataset//DSC_0031_NEF_embedded.axy"...
Field 1 did not solve (index index-4219.fits, field objects 1-10).
Field 1 did not solve (index index-4218.fits, field objects 1-10).
Field 1 did not solve (index index-4116.fits, field objects 1-10).
Field 1 did not solve (index index-4219.fits, field objects 11-20).
Field 1 did not solve (index index-4218.fits, field objects 11-20).
  log-odds ratio 246.665 (1.33452e+107), 46 match, 0 conflict, 61 distractors, 60 index.
  RA,Dec = (286.214,-4.13486), pixel scale 24.1215 arcsec/pix.
  Hit/miss:   Hit/miss: +++-+++++-+++++-++-++--+-++-++-++-++++++++-++-++-------++-+--------+--+--+--------------+-+---+-----
Field 1: solved with index index-4116.fits.
Field 1 solved: writing to file /home/roxo/unh-startracker-master/dataset//DSC_0031_NEF_embedded.solved to indicate this.
Field: /home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.jpg
Field center: (RA,Dec) = (286.209668, -4.122230) deg.
Field center: (RA H:M:S, Dec D:M:S) = (19:04:50.320, -04:07:20.030).
Field size: 25.6171 x 17.3108 degrees
Field rotation angle: up is -150.268 degrees E of N
Creating new FITS file "/home/roxo/unh-startracker-master/dataset//DSC_0031_NEF_embedded.new"...

wcsFile ========= DSC_0031_NEF_embedded.wcs 
command is ['/usr/local/astrometry/bin/plotann.py', '--brightcat', '/home/roxo/unh-startracker-master/astrometry1/catalogs/brightstars.fits', '/home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.wcs']
Traceback (most recent call last):
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 310, in <module>
    runCenterPixelMethod()  
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 293, in runCenterPixelMethod
    starList = getStarDict(wcsFile)
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 128, in getStarDict
    starDict = json.loads(starDict)
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```
the error appear on getStarDict whene loads json starDict!

**Comment by fsbr** [comment]
You may need to modify that part of the code.  the imgTime variable is
metadata that corresponded to my specific camera.

**Comment by masoudline** [comment]
ok but i test your demo image and these errors show and imgTime error debugged:
errors belongs to getStarDict function:
```
/usr/bin/python2.7 /home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py
ATTEMPTING TO ACCESS DATA DIR /home/roxo/unh-startracker-master/dataset/
['/home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.jpg']
Reading input file 1 of 1: "/home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.jpg"...
jpegtopnm: WRITING PPM FILE
Read file stdin: 3872 x 2592 pixels x 1 color(s); maxval 255
Using 8-bit output
Extracting sources...
Downsampling by 2...
simplexy: found 2197 sources.
Solving...
Reading file "/home/roxo/unh-startracker-master/dataset//DSC_0031_NEF_embedded.axy"...
Field 1 did not solve (index index-4219.fits, field objects 1-10).
Field 1 did not solve (index index-4218.fits, field objects 1-10).
Field 1 did not solve (index index-4116.fits, field objects 1-10).
Field 1 did not solve (index index-4219.fits, field objects 11-20).
Field 1 did not solve (index index-4218.fits, field objects 11-20).
  log-odds ratio 246.665 (1.33452e+107), 46 match, 0 conflict, 61 distractors, 60 index.
  RA,Dec = (286.214,-4.13486), pixel scale 24.1215 arcsec/pix.
  Hit/miss:   Hit/miss: +++-+++++-+++++-++-++--+-++-++-++-++++++++-++-++-------++-+--------+--+--+--------------+-+---+-----
Field 1: solved with index index-4116.fits.
Field 1 solved: writing to file /home/roxo/unh-startracker-master/dataset//DSC_0031_NEF_embedded.solved to indicate this.
Field: /home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.jpg
Field center: (RA,Dec) = (286.209668, -4.122230) deg.
Field center: (RA H:M:S, Dec D:M:S) = (19:04:50.320, -04:07:20.030).
Field size: 25.6171 x 17.3108 degrees
Field rotation angle: up is -150.268 degrees E of N
Creating new FITS file "/home/roxo/unh-startracker-master/dataset//DSC_0031_NEF_embedded.new"...

wcsFile ========= DSC_0031_NEF_embedded.wcs 
command is ['/usr/local/astrometry/bin/plotann.py', '--brightcat', '/home/roxo/unh-startracker-master/astrometry1/catalogs/brightstars.fits', '/home/roxo/unh-startracker-master/dataset/DSC_0031_NEF_embedded.wcs']
Traceback (most recent call last):
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 310, in <module>
    runCenterPixelMethod()  
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 293, in runCenterPixelMethod
    starList = getStarDict(wcsFile)
  File "/home/roxo/unh-startracker-master/analysisAndPlots/maps/CPM.py", line 128, in getStarDict
    starDict = json.loads(starDict)
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

**Comment by masoudline** [comment]
when I check errors and debugging those line to line understand that errors made by sdProc.stdout. because of these function return "none" object! but I can't solve it!

**Comment by fsbr** [comment]
Are you properly generating the WCS file

**Comment by fsbr** [comment]
Take a short break and return w fresh mind.

**Comment by masoudline** [comment]
yes this file generate correctly

**Comment by masoudline** [comment]
i change command=[] with command = " " and replace subproc.call  with os.Popen() but the new problem is Json data because output of above command is:
```
Bright-star catalog:  
Bright-star catalog: \u03b7 Ser 
Bright-star catalog: 59Ser 
Bright-star catalog: 60Ser 
Bright-star catalog: 61Ser 
Bright-star catalog: \u03b1 Sct 
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog: \u03b4 Sct 
Bright-star catalog: \u03b5 Sct 
Bright-star catalog: 4Aql 
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog: 5Aql 
Bright-star catalog: \u03b2 Sct 
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog: \u03b8 1Ser Alya
Bright-star catalog: \u03b8 2Ser 
Bright-star catalog: \u03b7 Sct 
Bright-star catalog: 64Ser 
Bright-star catalog:  
Bright-star catalog: 12Aql 
Bright-star catalog: 14Aql 
Bright-star catalog:  
Bright-star catalog: 15Aql 
Bright-star catalog: \u03bb Aql 
Bright-star catalog: 20Aql 
Bright-star catalog:  
Bright-star catalog: 21Aql 
Bright-star catalog: 23Aql 
Bright-star catalog: 26Aql 
Bright-star catalog: 27Aql 
Bright-star catalog: \u03c5 Sgr 
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog:  
Bright-star catalog: \u03bd Aql 
Bright-star catalog: 35Aql 
Bright-star catalog: 36Aql 
Bright-star catalog: 37Aql 
Bright-star catalog: \u03ba Aql 
Bright-star catalog: \u03b9 Aql 
Bright-star catalog:  
Bright-star catalog: 42Aql 
Bright-star catalog: 45Aql 
Bright-star catalog:  
[{"radius": 0.0, "names": ["\u03b7 Ser"], "pixelx": 3443.6442605851516, "pixely": 1955.096858341861, "vmag": 3.2000000000000002, "type": "bright"}, {"radius": 0.0, "names": ["59Ser"], "pixelx": 3484.7158520668045, "pixely": 1438.1923108581775, "vmag": 5.2000000000000002, "type": "bright"}, {"radius": 0.0, "names": ["60Ser"], "pixelx": 3236.2561747212826, "pixely": 1675.5427322848664, "vmag": 5.2999999999999998, "type": "bright"}, {"radius": 0.0, "names": ["61Ser"], "pixelx": 3236.1922381918957, "pixely": 1504.8895123054856, "vmag": 5.9000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["\u03b1 Sct"], "pixelx": 2581.883065346984, "pixely": 2382.1931224112436, "vmag": 3.7999999999999998, "type": "bright"}, {"radius": 0.0, "names": ["\u03b4 Sct"], "pixelx": 2293.8099436196067, "pixely": 2352.8040360952596, "vmag": 4.7000000000000002, "type": "bright"}, {"radius": 0.0, "names": ["\u03b5 Sct"], "pixelx": 2312.3543819277947, "pixely": 2228.281644846741, "vmag": 4.9000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["4Aql"], "pixelx": 3044.888688413234, "pixely": 865.1130536636905, "vmag": 5.0, "type": "bright"}, {"radius": 0.0, "names": ["5Aql"], "pixelx": 2764.1128492035195, "pixely": 1227.5633657497672, "vmag": 5.9000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["\u03b2 Sct"], "pixelx": 2458.839967346989, "pixely": 1703.6344338970125, "vmag": 4.2000000000000002, "type": "bright"}, {"radius": 0.0, "names": ["\u03b8 1Ser", "Alya"], "pixelx": 2834.461880076679, "pixely": 373.2421959237647, "vmag": 4.5999999999999996, "type": "bright"}, {"radius": 0.0, "names": ["\u03b8 2Ser"], "pixelx": 2833.5760362364654, "pixely": 373.037478059117, "vmag": 4.9000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["\u03b7 Sct"], "pixelx": 2059.287722730574, "pixely": 1661.9899228993918, "vmag": 4.7999999999999998, "type": "bright"}, {"radius": 0.0, "names": ["64Ser"], "pixelx": 2674.4641399644306, "pixely": 572.7798270850371, "vmag": 5.5, "type": "bright"}, {"radius": 0.0, "names": ["12Aql"], "pixelx": 1918.9400068110247, "pixely": 1563.1558220249367, "vmag": 4.0, "type": "bright"}, {"radius": 0.0, "names": ["14Aql"], "pixelx": 2029.8340739278194, "pixely": 1277.3200562148527, "vmag": 5.4000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["15Aql"], "pixelx": 1939.278860028278, "pixely": 1282.5473752674175, "vmag": 5.4000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["\u03bb Aql"], "pixelx": 1835.190150874494, "pixely": 1368.799411435231, "vmag": 3.3999999999999999, "type": "bright"}, {"radius": 0.0, "names": ["20Aql"], "pixelx": 1403.5467695661985, "pixely": 1647.1854888593548, "vmag": 5.2999999999999998, "type": "bright"}, {"radius": 0.0, "names": ["21Aql"], "pixelx": 2122.6149563587046, "pixely": 300.03840221317023, "vmag": 5.0999999999999996, "type": "bright"}, {"radius": 0.0, "names": ["23Aql"], "pixelx": 1876.2965399064262, "pixely": 368.46844881539425, "vmag": 5.0999999999999996, "type": "bright"}, {"radius": 0.0, "names": ["26Aql"], "pixelx": 1334.6299770600701, "pixely": 1176.7622061501784, "vmag": 5.0, "type": "bright"}, {"radius": 0.0, "names": ["27Aql"], "pixelx": 1664.1331911454054, "pixely": 588.4171305007536, "vmag": 5.4000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["\u03c5 Sgr"], "pixelx": 513.7224167097368, "pixely": 2544.5971052011946, "vmag": 4.5999999999999996, "type": "bright"}, {"radius": 0.0, "names": ["\u03bd Aql"], "pixelx": 1561.4970836724833, "pixely": 317.7042970728021, "vmag": 4.5999999999999996, "type": "bright"}, {"radius": 0.0, "names": ["35Aql"], "pixelx": 1598.4768461166705, "pixely": 59.00849180774685, "vmag": 5.7999999999999998, "type": "bright"}, {"radius": 0.0, "names": ["36Aql"], "pixelx": 1196.997848151937, "pixely": 649.7100174307371, "vmag": 5.0, "type": "bright"}, {"radius": 0.0, "names": ["37Aql"], "pixelx": 481.6265327907606, "pixely": 1588.0307785676534, "vmag": 5.0999999999999996, "type": "bright"}, {"radius": 0.0, "names": ["\u03ba Aql"], "pixelx": 682.780452440767, "pixely": 1090.5452385264598, "vmag": 4.9000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["\u03b9 Aql"], "pixelx": 1107.614073575668, "pixely": 340.4156415351082, "vmag": 4.2999999999999998, "type": "bright"}, {"radius": 0.0, "names": ["42Aql"], "pixelx": 826.9737175890116, "pixely": 761.8136745824778, "vmag": 5.4000000000000004, "type": "bright"}, {"radius": 0.0, "names": ["45Aql"], "pixelx": 1023.9858204245359, "pixely": 177.68786487623254, "vmag": 5.5999999999999996, "type": "bright"}]
```
so last line is json and other line is not.

**Comment by fsbr** [comment]
That output looks correct. If you publish research please cite my thesis

**Comment by masoudline** [comment]
how can i access to estimate navigation az an ze for observation. i want use these code for a boat

**Comment by masoudline** [comment]
please can you tell me whats these output? i can run CPM.py and these output show:
what are those?
```
actually used observations are ===== [{'dec': [-3, 41.9393], u'vmag': 5.4, 'yOff': 18.679943785147316, 'sha': [74, 16.375], 'newx': 1903.1742107788805, 'newy': 1459.349449965055, u'radius': 0.0, u'names': [u'14Aql'], u'pixelx': 2029.8340739278194, u'pixely': 1277.3200562148527, 'xOff': 93.83407392781942, u'type': u'bright', 'ho': 57.61605798426531}, {'dec': [-4, 52.95334], u'vmag': 5.4, 'yOff': 13.452624732582535, 'sha': [73, 26.2652], 'newx': 1812.5232861892948, 'newy': 1456.1900987781364, u'radius': 0.0, u'names': [u'15Aql'], u'pixelx': 1939.278860028278, u'pixely': 1282.5473752674175, 'xOff': 3.2788600282779043, u'type': u'bright', 'ho': 57.52521908651406}, {'dec': [-4, 52.95335], u'vmag': 3.4, 'yOff': -72.79941143523092, 'sha': [73.0, 26.26254], 'newx': 1700.9127026615713, 'newy': 1532.4590358429462, u'radius': 0.0, u'names': [u'\u03bb Aql'], u'pixelx': 1835.190150874494, u'pixely': 1368.799411435231, 'xOff': -100.80984912550593, u'type': u'bright', 'ho': 56.885915146956215}]
OBSERVATIONS ====== [<ST.Observation instance at 0x7fea84cee290>, <ST.Observation instance at 0x7fea84cee368>, <ST.Observation instance at 0x7fea84cee440>]
hoilist =[57.61605798426531, 57.52521908651406, 56.885915146956215]
hciList=[29.289340634862935, 28.781501171660526, 28.7815256816803]

Process finished with exit code 0
```

and how can i estimate my position from these output. 
special thanks for your answer.

**Comment by fsbr** [comment]
There should be a call to ST.py that performs the calculation

**Comment by masoudline** [comment]
yes but whats these variable?
holists
hcilist?
and what number is my position?
thanks

**Comment by masoudline** [comment]
And how can you defined indez-xx number for solve-field astrometry?

My project dont connect to internet and this problem exists that how can i understand what index suitable fof processing.

**Comment by fsbr** [comment]
Can you rephrase the question

**Comment by masoudline** [comment]
1) what these variables:

holists
hcilist?

2) what number is my position?

3) I want to build a celestial navigation system and use it on my boat. so I will be used a camera and take a picture from the night sky, I can't connect to internet and nova.astrometry.net website for search index-xx number and download it!
how can I estimate index-xx number for search on a taken picture from camera!???

**Comment by fsbr** [comment]
The latitude and longitude are Lf and Bf. Please read the index of the
nautical almanac which explains this.

**Comment by masoudline** [comment]
Thanks for your answer 
Please send your answer about Question 2 and 3?

**Comment by fsbr** [comment]
You can download the astrometry program to run locally on your computer
from astrometry.net. You need to read the algorithm from the nautical
almanac which will explain the ideas behind the celestial navigation. There
is an example calculation in the back of the book. Please attempt these
problems and you will understand more.

On Oct 12, 2017 2:15 PM, "masoudline" <notifications@github.com> wrote:

Thanks for your answer
Please send your answer about Question 2 and 3?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/fsbr/unh-startracker/issues/1#issuecomment-336221691>,
or mute the thread

**Comment by masoudline** [comment]
did Your project estimate observer position from a taken picture of night sky?

**Comment by fsbr** [comment]
Yes. Please read the thesis hosted on my github to learn more
